scipy/special/tests/test_basic.py

```diff
@@ -2642,6 +2642,12 @@ def test_hankl2e(self):
         hankrl2e = special.hankel2e(1,.1)
         assert_almost_equal(hank2e,hankrl2e,8)
 
+    def test_hankel2_gh4517(self):
+        # Test edge case reported in https://github.com/scipy/scipy/issues/4517
+        res = special.hankel2(0, 0)
```

Comment:
Poking around a bit:

```python
>>> import mpmath as mp
>>> mp.hankel2(0, 0)
mpc(real='nan', imag='+inf')
```

Wolfram Alpha says: `Hankel H2(0,0)` -> "complex infinity"

Pauli indicated in the matching issue that `nan+infj` is a valid representation of complex infinity.

The current patch is obviously better for the imaginary component of the number, but the real component seems more confusing relative to other sources? Are `1+infj` and `nan+infj` equally valid representations of complex infinity? I was under the impression complex infinity couldn't have a finite real component, but I'm definitely not an expert.

Reply:
> I was under the impression complex infinity couldn't have a finite real component, but I'm definitely not an expert.

Having one finite component is fine. See for instance the *Error handing and special values* section of the `clog` documentation [here](https://en.cppreference.com/w/c/numeric/complex/clog).

![image](https://github.com/user-attachments/assets/a42cb68c-09f1-451c-985a-df1ec41dcae6)